images.CI/linux-and-win/convert-to-vhd.ps1

```diff
@@ -0,0 +1,211 @@
+param (
+  [String] [Parameter (Mandatory = $true)] $SubscriptionId,
+  [String] [Parameter (Mandatory = $true)] $Location,
+  [String] [Parameter (Mandatory = $true)] $ResourceGroupName,
+  
+  [String] [Parameter (Mandatory = $true)] $ManagedImageName,
+  
+  [String] [Parameter (Mandatory = $true)] $GalleryName,
+  [String] [Parameter (Mandatory = $true)] $GalleryImageSku,
+  [String] [Parameter (Mandatory = $true)] $GalleryImageVersion,
+  
+  [String] [Parameter (Mandatory = $true)] $StorageAccountName,
+  [String] [Parameter (Mandatory = $true)] $StorageAccountContainerName,
+  [String] [Parameter (Mandatory = $true)] $VhdName,
+  
+  [String] [Parameter (Mandatory = $true)] $ClientId,
+  [String] [Parameter (Mandatory = $true)] $ClientSecret,
+  [String] [Parameter (Mandatory = $true)] $TenantId
+)
+
+function Get-DestinationBlobUri {
+  param (
+    [string] [Parameter(Mandatory = $true)] $SubscriptionId,
+
+    [string] [Parameter(Mandatory = $true)] $ResourceGroupName,
+    [string] [Parameter(Mandatory = $true)] $StorageAccountName,
+    [string] [Parameter(Mandatory = $true)] $ContainerName,
+    [string] [Parameter(Mandatory = $true)] $ImageBlobName,
+    
+    [string] [Parameter(Mandatory = $true)] $TenantId
+  )
+
+  Write-Host "List the access keys or Kerberos keys for a $StorageAccountName storage account."
```

Comment:
I we probably can use SP rather than sas token to avoid security concerns as we had in prod, at least worth checking

Reply:
It will overcomplicate our CI. Our SPN has no certificate authorization, so firstly we need to create a cert for it, store it in some KV, etc. 
To simplify the solution we can use `az storage blob copy start` command, by login to Azure using OIDC.